EmployeeLeaveTracking/EmployeeLeaveTracking.Services/Services/LeaveRequestService.cs

```diff
@@ -142,17 +144,16 @@ public NewLeaveRequestDTO CreateNewLeaveRequest(NewLeaveRequestDTO leaveRequest)
             return leaveRequest;
         }
 
-
-
         public NewLeaveRequestDTO NewCreateNewLeaveRequest(NewLeaveRequestDTO leaveRequest)
         {
-            //checking if there are any existing leave requests for the same user and overlapping dates
+            //checking if there are any existing approved leave requests for the same user and overlapping dates
             bool isLeaveAlreadyExists = _dbContext.LeaveRequests
                 .Any(lr => lr.EmployeeId == leaveRequest.EmployeeId &&
                            lr.StartDate <= leaveRequest.EndDate &&
                            lr.EndDate >= leaveRequest.StartDate &&
```

Comment:
There is a logical issue in the query as per my understanding.
What if Emp has applied for a leave on 30th July 
and if he/she tries to again apply for a leave from 30th July to 5th Aug

in this case start date for 1st request is 30th and End date is 30th
where as in second case start date is 30th and End date is 5th

By your logic I believe (lr.StartDate <= leaveRequest.EndDate && lr.EndDate >= leaveRequest.StartDate ) might not work. I am not 100% sure but I would have validated this